test/rule_definition_tools.jl

```diff
@@ -0,0 +1,65 @@
+@testset "rule_definition_tools.jl" begin
+    @testset "@non_differentiable" begin
+        @testset "nondiff_2_1" begin
+            nondiff_2_1(x, y) = fill(7.5, 100)[x + y]
+            @non_differentiable nondiff_2_1(::Any, ::Any)
+            @test frule((Zero(), 1.2, 2.3), nondiff_2_1, 3, 2) == (7.5, DoesNotExist())
+            res, pullback = rrule(nondiff_2_1, 3, 2)
+            @test res == 7.5
+            @test pullback(4.5) == (NO_FIELDS, DoesNotExist(), DoesNotExist())
+        end
+
+        @testset "nondiff_1_2" begin
```

Comment:
Could you explain why this testset / function is called nondiff_2_1?

Reply:
```suggestion
        @testset "one input, 2-tuple output function" begin
```